src/Temporalio/Worker/WorkflowWorkerOptions.cs

```diff
@@ -22,5 +22,6 @@ internal record WorkflowWorkerOptions(
         Action<WorkflowInstance, Exception?> OnTaskCompleted,
         Lazy<MetricMeter> RuntimeMetricMeter,
         IReadOnlyCollection<Type>? WorkerLevelFailureExceptionTypes,
-        bool DisableEagerActivityExecution);
+        bool DisableEagerActivityExecution,
+        Action<string> AssertValidActivity);
```

Comment:
Pedantic, but I'd call this `AssertValidLocalActivity` (don't need to change the source method in activity worker, since that makes sense). The concern is that non-local activities don't use this so the name may be a bit confusing.